inc/azure_c_util/interlocked.h

```diff
@@ -18,42 +18,46 @@
 extern "C" {
 #endif
 
+#ifdef _WIN32
+#define volatile_atomic volatile
```

Comment:
>volatile_atomic [](start = 8, length = 15)

I know I told you to add this and we need something like that. I'm not sure if we should maybe rename it to something else though that has "interlocked" in the name (so it doesn't collide with anything else) or not (because the current name says exactly what it is, which is nice)... Dan or Andrei might have an opinion about this.

But I think it's fine for now as it's a simple "find and replace" rename. You can probably just leave it for now as we do need to declare the types as `_Atomic` to use C11 atomics from what I understand.